share/doc/wake/quickref.md

```diff
@@ -220,8 +220,31 @@ global def runit _ =
+* The `pre` argument is of type `Result RunnerInput Error → Pair (Result RunnerInput Error) a`. The `RunnerInput` tuple is a subset of the Plan tuple, serving to restrict the fields that the runner can access. The `pre` function is called before the job is run, allowing the runner to modify the input to provide the requested resources. For example, the RISC-V runner would run `runnerInput | editRunnerInputEnvironment addRISCVEnvironment` where `addRISCVEnvironment` is a function that sets the RISCV environment variable. The return value of `pre` is the modified `RunnerInput` paired with a free `a` type that can contain any sideband data that the `pre` function needs to provide to the `post` function.
+* The `post` argument is of type `Pair (Result RunnerOutput Error) a → Result RunnerOutput Error` and is similar to the `pre` function but is called after the job has run. `post` is for editing the reported outputs/inputs/usage of the job. For example, the fuse runner uses the `post` hook to set the output and input files reported by the FUSE filesystem. 
+* The last argument is the base runner that the current runner is built on top of. This is because all runners must be built on top of a preexisting runner. For example, in environment-example-sifive, localRISCVRunner and defaultRISCVRunner are built on top of localRunner and defaultRunner respectively.
+* `publish runner` must be executed for Wake to recognize a runner
+
+Runners are chosen for a given Plan based on the value returned by their `score` function and the plan's `RunnerFilter` field.
+
+* In `runJob`,  `subscribe runner` is called to retrieve the list of all published runners. The `RunnerFilter` function inside the Plan tuple can be set to filter out runners that the plan wants to exclude from consideration. Then the `score` function of each runner is called and runners that return Fail or a score <= 0.0 are excluded. Of the remaining runners, the one with the highest score is picked. 
+* Local runners can only run when the Plan tuple has `LocalOnly` set to true
+* Default runners can only run when the Plan tuple has `LocalOnly` set to false
+
+#### Using environment packages 
+
+Environment packages can be added to a workspace to provide tools (using runners) for running the workflow. You can check out [environment-example-sifive](https://github.com/sifive/environment-example-sifive) as an example of an environment package that contains runners for Wake-based workflows. 
```

Comment:
It might be nice to explicitly mention that an environment package is in fact a collection of Wake files which define and publish runners. It's also probably a good idea to mention that environment packages are meant to be defined for each unique running environment, which decouples the Wake rules from how tools are specifically installed in a specific environment.